bundles/org.openhab.core.config.core/src/main/java/org/eclipse/smarthome/config/core/Configuration.java

```diff
@@ -59,29 +62,21 @@ public Configuration(final @Nullable Configuration configuration) {
      * @param properties the properties the configuration should be filled. If null, an empty configuration is created.
      */
     public Configuration(Map<String, Object> properties) {
-        this(properties, false);
+        this(properties == null ? emptyMap() : properties, false);
     }
 
     /**
      * Create a new configuration.
      *
-     * @param properties the properties to initialize (may be null)
+     * @param properties        the properties to initialize (may be null)
      * @param alreadyNormalized flag if the properties are already normalized
      */
     private Configuration(final @Nullable Map<String, Object> properties, final boolean alreadyNormalized) {
-        if (properties == null) {
-            this.properties = new HashMap<>();
-        } else {
-            if (alreadyNormalized) {
-                this.properties = new HashMap<>(properties);
-            } else {
-                this.properties = ConfigUtil.normalizeTypes(properties);
-            }
-        }
+        this.properties = synchronizedMap(alreadyNormalized ? new HashMap<>(properties) : normalizeTypes(properties));
```

Comment:
Why has this been changed?
The argument `properties` is marked nullable.
I am pretty sure `new HashMap<>(properties)` will throw an exception if it is called with a null argument.

Reply:
I forgot to remove the nullable annotation - it's no longer possible for the properties parameter to be null.

Reply:
In essence, I simplified this (private) constructor to accept only nonnull properties and made sure that all the other, public constructors call it with nonnull arguments.

Reply:
As this constructor is private, I am fine with the changes if the nullable annotation is removed and so `properties` is defined to be non null.

Reply:
I removed the nullable annotation.